src/controllers/__tests__/userDataProcessingController.test.ts

```diff
@@ -0,0 +1,269 @@
+    const apiClient = new ApiClient("XUZTCT88A51Y311X", "");
+    const userDataProcessingService = new UserDataProcessingService(apiClient);
+    const controller = new UserDataProcessingController(
+      userDataProcessingService
+    );
+
+    const response = await controller.getUserDataProcessing(req);
+    response.apply(res);
+
+    // getUserDataProcessing is not called
+    expect(mockGetUserDataProcessing).not.toBeCalled();
+    expect(res.json).toHaveBeenCalledWith(badRequestErrorResponse);
+  });
+
+  it("should return a ResponseErrorNotFound if no user data processing was found", async () => {
+    const req = mockReq();
+    const res = mockRes();
+
+    mockGetUserDataProcessing.mockReturnValue(
+      Promise.resolve(
+        ResponseErrorNotFound("Not Found", "User data processing not found")
+      )
+    );
+
+    req.user = mockedUser;
+    req.params = { choice: aUserDataProcessingChoice };
+
+    const apiClient = new ApiClient("XUZTCT88A51Y311X", "");
+    const userDataProcessingService = new UserDataProcessingService(apiClient);
+    const controller = new UserDataProcessingController(
+      userDataProcessingService
+    );
+    const response = await controller.getUserDataProcessing(req);
+    response.apply(res);
+
+    expect(mockGetUserDataProcessing).toHaveBeenCalledWith(
+      mockedUser,
+      aUserDataProcessingChoice
+    );
+    expect(response).toEqual({
+      ...userDataProcessingMissingErrorResponse,
+      apply: expect.any(Function)
+    });
+  });
+});
+
+describe("UserDataProcessingController#upsertUserDataProcessing", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should return a valid upsertedUserDataProcessingby calling UserDataProcessingService with valid values", async () => {
+    const req = mockReq();
+
+    mockUpsertUserDataProcessing.mockReturnValue(
+      Promise.resolve(ResponseSuccessJson(aUserDataProcessingResponse))
+    );
+
+    req.user = mockedUser;
+    req.body = mockedUserDataProcessingChoiceRequest;
+
+    const apiClient = new ApiClient("XUZTCT88A51Y311X", "");
+    const userDataProcessingService = new UserDataProcessingService(apiClient);
+    const controller = new UserDataProcessingController(
+      userDataProcessingService
+    );
+    const response = await controller.upsertUserDataProcessing(req);
+
+    const errorOrUserDataProcessingChoice = UserDataProcessingChoiceRequest.decode(
+      req.body
+    );
+    expect(isRight(errorOrUserDataProcessingChoice)).toBeTruthy();
+
+    expect(mockUpsertUserDataProcessing).toHaveBeenCalledWith(
+      mockedUser,
+      errorOrUserDataProcessingChoice.value
+    );
+    expect(response).toEqual({
+      apply: expect.any(Function),
+      kind: "IResponseSuccessJson",
+      value: aUserDataProcessingResponse
+    });
+  });
+
+  it("should return a BadRequestError response by calling upsertUserDataProcessing on the UserDataProcessingService with empty user and valid upsert user", async () => {
```

Comment:
```suggestion
  it("should return a bad request response if request user is empty in the request", async () => {
```

the fact that we're calling `UserDataProcessingController#upsertUserDataProcessing` is inferred by the describe call